### PR TITLE
docs: add remote-per-output

### PR DESCRIPTION
**Pull request by efiop**
Per https://github.com/iterative/dvc/pull/6486

### CODE REVIEW

File: content/docs/user-guide/project-structure/dvc-files.md [hunk 1]
```diff
@@ -65,6 +66,7 @@ The following subfields may be present under `outs` entries:
...
 | `persist`                       | Whether the output file/dir should remain in place while `dvc repro` runs (`false` by default: outputs are deleted when `dvc repro` starts)                                                                                                                                                                                           |
 | `desc`                          | (Optional) user description for this output (supported in metrics and plots too). This doesn't affect any DVC operations.                                                                                                                                                                                                             |
+| `remote`                        | (Optional) name of the remote to use for pushing/fetching                                                                                                                                                                                                                                                                             |
```
Comment (jorgeorpinel) [inline content/docs/user-guide/project-structure/dvc-files.md:69]: Just moving this above less interesting fields...
```suggestion
| `remote`                        | (Optional) name of the remote to use for pushing/fetching                                                                                                                                                                                                                                                                             |
| `persist`                       | Whether the output file/dir should remain in place while `dvc repro` runs (`false` by default: outputs are deleted when `dvc repro` starts)                                                                                                                                                                                           |
| `desc`                          | (Optional) user description for this output (supported in metrics and plots too). This doesn't affect any DVC operations.                                                                                                                                                                                                             |
```

File: content/docs/user-guide/project-structure/pipelines-files.md [hunk 2]
```diff
@@ -385,6 +385,7 @@ validation and auto-completion.
 | `persist`    | Whether the output file/dir should remain in place while `dvc repro` runs (`false` by default: outputs are deleted when `dvc repro` starts                                                                                                                                         |
 | `desc`       | (Optional) user description for this output. This doesn't affect any DVC operations.                                                                                                                                                                                               |
 | `checkpoint` | Set to `true` to let DVC know that this output is associated with [in-code checkpoints](/doc/user-guide/experiment-management#checkpoints-in-source-code). These outputs are reverted to their last cached version at `dvc exp run` and also `persist` during the stage execution. |
+| `remote`     | (Optional) name of the remote to use for pushing/fetching.                                                                                                                                                                                                                         |
```
Comment (jorgeorpinel) [inline content/docs/user-guide/project-structure/pipelines-files.md:388]: Same
```suggestion
| `remote`     | (Optional) name of the remote to use for pushing/fetching.                                                                                                                                                                                                                         |
| `persist`    | Whether the output file/dir should remain in place while `dvc repro` runs (`false` by default: outputs are deleted when `dvc repro` starts                                                                                                                                         |
| `checkpoint` | Set to `true` to let DVC know that this output is associated with [in-code checkpoints](/doc/user-guide/experiment-management#checkpoints-in-source-code). These outputs are reverted to their last cached version at `dvc exp run` and also `persist` during the stage execution. |
| `desc`       | (Optional) user description for this output. This doesn't affect any DVC operations.                                                                                                                                                                                               |
```

File: content/docs/user-guide/project-structure/dvc-files.md (hunk 1 above)
Comment (jorgeorpinel) [reply content/docs/user-guide/project-structure/dvc-files.md:69]: p.s. not sure we need the "(optional)" remarks. They're all optional I think. Maybe they should all indicate what's the default value instead. Out of scope though (but feel free to apply that rule to the one you added).

File: content/docs/command-reference/fetch.md [hunk 3]
```diff
@@ -60,9 +60,13 @@ directories), `.dvc` files, and stage names (found in `dvc.yaml`).
...
-The default remote is used (see
-[`dvc config core.remote`](/doc/command-reference/config#core)) unless a
-specific one is given with `--remote`.
+Order of priority of remotes:
+
+1. `remote` specified in a corresponding dvcfile for particular output.
+2. `--remote` CLI option.
+3. `core.remote`config option (see `dvc remote default`).
+
+See `dvc remote` for more information on how to configure remotes.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/fetch.md:69]: Feel free to apply this copy edit to the other instances:
```suggestion
The `dvc remote` used is determined in order, based on

1. the `remote` fields in the `dvc.yaml` or `.dvc` files.
2. the value passed to the `--remote` option via CLI.
3. the value of the `core.remote` config option (see `dvc remote default`).
```

**Comment by jorgeorpinel** [comment]
An example would be nice, maybe in https://dvc.org/doc/command-reference/remote. Also maybe mention something about this in https://dvc.org/doc/command-reference/remote/default?

**Comment by shcheklein** [comment]
@jorgeorpinel can we apply quick fixes and merge?

File: content/docs/command-reference/pull.md [hunk 4]
```diff
@@ -53,9 +53,13 @@ project's cache                  ++ | dvc pull |
...
-The default remote is used (see `dvc remote default`) unless a specific one is
-given with `--remote`. See `dvc remote` for more information on how to configure
-a remote.
+Order of priority of remotes:
+
+1. `remote` specified in a corresponding dvcfile for particular output.
+2. `--remote` CLI option.
+3. `core.remote`config option (see `dvc remote default`).
+
+See `dvc remote` for more information on how to configure remotes.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/pull.md:62]:
```suggestion
The `dvc remote` used is determined in order, based on

1. the `remote` fields in the `dvc.yaml` or `.dvc` files.
2. the value passed to the `--remote` option via CLI.
3. the value of the `core.remote` config option (see `dvc remote default`).
```

File: content/docs/command-reference/push.md [hunk 5]
```diff
@@ -33,9 +33,13 @@ use cases for these commands.
...
-The default remote is used (see `dvc remote default`) unless a specific one is
-given with `--remote`. See `dvc remote` for more information on how to configure
-them.
+Order of priority of remotes:
+
+1. `remote` specified in a corresponding dvcfile for particular output.
+2. `--remote` CLI option.
+3. `core.remote`config option (see `dvc remote default`).
+
+See `dvc remote` for more information on how to configure remotes.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/push.md:42]:
```suggestion
The `dvc remote` used is determined in order, based on

1. the `remote` fields in the `dvc.yaml` or `.dvc` files.
2. the value passed to the `--remote` option via CLI.
3. the value of the `core.remote` config option (see `dvc remote default`).
```

File: content/docs/command-reference/status.md [hunk 6]
```diff
@@ -106,6 +106,14 @@ that.
   the `--remote` option is not used, DVC will compare against the default remote
   (specified in the `core.remote` config option).
 
+  Order of priority of remotes:
+
+  1. `remote` specified in a corresponding dvcfile for particular output.
+  2. `--remote` CLI option.
+  3. `core.remote`config option (see `dvc remote default`).
+
+  See `dvc remote` for more information on how to configure remotes.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/status.md:115]:
```suggestion
The `dvc remote` used is determined in order, based on

1. the `remote` fields in the `dvc.yaml` or `.dvc` files.
2. the value passed to the `--remote` option via CLI.
3. the value of the `core.remote` config option (see `dvc remote default`).
```

**Review by jorgeorpinel — APPROVED** [review]
> can we apply quick fixes

Done @shcheklein, this is mergeable now I think.

I still think an example would be nice (an other small details mentioned above) but I can't do that right now myself.

**Comment by jorgeorpinel** [comment]
Unrelated to this PR but we need to figure out what's wrong with Restyled. It's been passing every single time recently even if the formatting [checks fail](https://app.circleci.com/jobs/github/iterative/dvc.org/10340?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link)... Maybe related with https://github.com/iterative/dvc.org/issues/2641 ⏳ 

image

**Review by jorgeorpinel — APPROVED** [review]
Fixed the formatting now! LGTM?

> Will address Restyled problems in https://github.com/iterative/dvc.org/pull/2774

**Review by shcheklein — APPROVED** [review]

**Comment by jorgeorpinel** [comment]
OK merging this. If you have a good example to add to https://dvc.org/doc/command-reference/remote please open another PR @efiop. And thanks!

**Pull request merged**